core/src/main/scala/org/apache/spark/h2o/backends/external/RestApiUtils.scala

```diff
@@ -19,22 +19,53 @@ package org.apache.spark.h2o.backends.external
 
 import java.io.File
 import java.net.URI
-import java.text.SimpleDateFormat
+import java.text.{MessageFormat, SimpleDateFormat}
 import java.util.Date
 
-import ai.h2o.sparkling.frame.{H2OChunk, H2OColumn, H2OFrame}
+import ai.h2o.sparkling.frame.{H2OChunk, H2OColumn, H2OColumnType, H2OFrame}
 import org.apache.http.client.utils.URIBuilder
-import org.apache.spark.h2o.H2OConf
+import org.apache.spark.h2o.{H2OConf, H2OContext}
 import org.apache.spark.h2o.utils.NodeDesc
 import water.api.schemas3.FrameChunksV3.FrameChunkV3
 import water.api.schemas3.FrameV3.ColV3
 import water.api.schemas3._
 
 trait RestApiUtils extends RestCommunication {
 
+  def isRestAPIBased(hc: Option[H2OContext] = None): Boolean = {
+    hc.getOrElse(H2OContext.ensure()).getConf.get("spark.ext.h2o.rest.api.based.client", "false") == "true"
```

Comment:
Maybe we could make `H2OContext.isRestAPIBased` public or package private and reuse it here.

Reply:
Good idea, I have removed the method on H2OContext and used the one from RestAPIUtils